### ISSUE TITLE
SWIFT_ACTIVE_COMPILATION_CONDITIONS are ignored

### ISSUE THREAD

**Issue by mrgrauel** [CocoaPods/Xcodeproj#478]
In version 1.4.3 are `SWIFT_ACTIVE_COMPILATION_CONDITIONS` ignored, which causes various CocoaPods to fail after an `pod update`.

Example: https://github.com/delba/Permission

**Comment by orta** [comment]
could be related to https://github.com/CocoaPods/Xcodeproj/issues/477

**Comment by louisdh** [comment]
I've tested `pod install` and `pod update` with a newly created project (by Xcode 8.3) and the following `podfile`:
```
target 'Test Cocoapods' do
  use_frameworks!

  pod 'Permission/Camera'
  pod 'Permission/Notifications'

end
```
Using the following gem versions:
`xcodeproj 1.4.3`
`cocoapods 1.2.0`

All works fine (no errors, Xcode build succeeds).
@mrgrauel Can you share your `podfile` and error(s) you're seeing?

**Comment by mrgrauel** [comment]
Try to use a Camera or Notification PermissionType. It's not available cause it was never compiled, cause the `SWIFT_ACTIVE_COMPILATION_CONDITIONS` attribute is ignored. It is used internal to determine which source should compile.

**Comment by louisdh** [comment]
The issue seems to be caused by e61685641d481bf2b0fa4a0e6584d334290c3bb8, where `SWIFT_ACTIVE_COMPILATION_CONDITIONS` was set to `DEBUG` (for debug Swift). Adding `$(inherited)` should fix this.

If you build in release mode the `SWIFT_ACTIVE_COMPILATION_CONDITIONS` are inherited correctly and the PermissionTypes are compiled correctly.

**Comment by mrgrauel** [comment]
Why is it set anyhow? Shouldn't the project decide which flags are set for this option?

**Comment by louisdh** [comment]
In Xcode 8 Apple has added a default `DEBUG` condition for Swift debug builds.

**Comment by louisdh** [comment]
#479 should fix this issue.

**Comment by louisdh** [comment]
This should be fixed by CocoaPods/CocoaPods#6625, which is now merged.
